migrations/m150610_162817_oauth.php

```diff
@@ -22,6 +22,7 @@ public function safeUp()
                 'client_id' => Schema::TYPE_STRING . '(80) NOT NULL',
                 'client_secret' => Schema::TYPE_STRING . '(80) NOT NULL',
                 'redirect_uri' => Schema::TYPE_TEXT . ' NOT NULL',
+                'user_id' => Schema::TYPE_INTEGER . ' NOT NULL',
```

Comment:
Upon saving token and refresh token model. you will face error because of the missing user_id. What I understand from user_id is the user identity id and should be linked with the client table. 
By this modification. Each time we create a token for a user and we need the user id we can take it from client table.